core/src/main/java/com/linecorp/armeria/client/UserClient.java

```diff
@@ -140,7 +140,8 @@ protected final O execute(@Nullable EventLoop eventLoop, Endpoint endpoint,
                               I req, BiFunction<ClientRequestContext, Throwable, O> fallback) {
         final DefaultClientRequestContext ctx;
         if (eventLoop == null) {
-            final ReleasableHolder<EventLoop> releasableEventLoop = factory().acquireEventLoop(endpoint);
+            final ReleasableHolder<EventLoop> releasableEventLoop = factory().acquireEventLoop(endpoint,
+                                                                                               sessionProtocol);
```

Comment:
How about acquiring an `EventLoop` after endpoint selection occurred, so that event loop is always chosen by a host `Endpoint`? i.e. Modify `DefaultClientRequestContext.init()` to accept a `ClientFactory` and call `acquireEventLoop(...)`.

Reply:
Yes, I will do that.
As discussed, we do not have to provide a way to set the maximum event loops for a group, will make it take always a host `Endpoint`. Thanks!